discussions-server/src/main/java/org/xwiki/contrib/discussions/internal/server/DefaultDiscussionMessageRequestCreator.java

```diff
@@ -0,0 +1,184 @@
+                "The discussion reference has not been provided.");
+        } else {
+            DiscussionReference discussionReference =
+                this.discussionReferencesResolver.resolve(serializedReference, DiscussionReference.class);
+            Optional<Discussion> discussionOptional = this.discussionService.get(discussionReference);
+            if (!discussionOptional.isPresent()) {
+                throw new DiscussionServerException(HttpServletResponse.SC_NOT_FOUND,
+                    String.format("Cannot find discussion with reference [%s]", serializedReference));
+            } else {
+                return this.createMessage(discussionOptional.get(), request);
+            }
+        }
+    }
+
+    private Optional<Message> createMessage(Discussion discussion, HttpServletRequest request)
+    {
+        String content = getContent(request);
+        Syntax syntax = getSyntax(request);
+        EditForm editForm = this.prepareForm(request);
+        DiscussionStoreConfigurationParameters parameters = new DiscussionStoreConfigurationParameters();
+        request.getParameterMap().forEach((key, value) -> {
+            if (key.startsWith(STORE_CONFIGURATION_PARAMETER_PREFIX)) {
+                String parameterKey = key.substring(STORE_CONFIGURATION_PARAMETER_PREFIX.length());
+                parameters.put(parameterKey, value);
+            }
+        });
+
+        // Handle temporary uploads
+        List<String> temporaryUploadedFiles = editForm.getTemporaryUploadedFiles();
+        parameters.put(TEMPORARY_UPLOADED_ATTACHMENTS, temporaryUploadedFiles);
+        Optional<Message> messageOptional;
+
+        String serializedReplyTo = request.getParameter(REPLY_TO_PARAMETER);
+        DocumentReference author = this.contextProvider.get().getAuthorReference();
+        String serializedAuthorReference = this.entityReferenceSerializer.serialize(author);
+        ActorReference actorReference = new ActorReference("user", serializedAuthorReference);
+
+        Message replyToMessage = null;
+        if (!StringUtils.isEmpty(serializedReplyTo)) {
+            MessageReference replyTo =
+                this.discussionReferencesResolver.resolve(serializedReplyTo, MessageReference.class);
+            Optional<Message> replyToReference = this.messageService.getByReference(replyTo);
+            if (replyToReference.isPresent()) {
+                replyToMessage = replyToReference.get();
+            }
+        }
+
+        if (replyToMessage == null) {
+            messageOptional = this.messageService
+                .create(content, syntax, discussion.getReference(), actorReference, true, parameters);
+        } else {
+            messageOptional = this.messageService
+                .createReplyTo(content, syntax, replyToMessage, actorReference, true, parameters);
+        }
+        return messageOptional;
+    }
+
+    private String getContent(HttpServletRequest request)
```

Comment:
Can be simplified using `RequestParameterConverter` but only on XWiki 13.5RC1+, should we consider upgrading the parent version?

Reply:
Well part of this PR is about upgrading the parent version to 14.10 :) Here it's because I just copy/pasted what was already existing, but I agree I should use `RequestParameterConverter` I'll check for it.